exp/api/v1alpha3/webhooks.go

```diff
@@ -0,0 +1,35 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+func aggregateObjErrors(gk schema.GroupKind, name string, allErrs field.ErrorList) error {
```

Comment:
I think we should be also refactoring all of the webhooks to use kerrors.NewAggregate https://github.com/kubernetes/apimachinery/blob/master/pkg/util/errors/errors.go#L46. Can then remove the length check as that's already implemented in kerrors.

Reply:
You mean to get rid of this `aggregateObjErrors` and use `kerrors.NewAggregate` like it's being done here https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/0c45468b245feb28565bba57938511c3cf3f5f59/exp/api/v1alpha3/azuremachinepool_webhook.go#L92

Reply:
Well, you don't even need to do
```
if len(errs) > 0 {
		return kerrors.NewAggregate(errs)
	}
```
as NewAggregate does that for you.

Reply:
Agree..made some code changes.PTAL